awx/api/serializers.py

```
@@ -2270,6 +2291,42 @@ class Meta:
         model = Group
 
 
+_VALID_PROXY_SCHEMES = ('http', 'https', 'socks', 'socks4', 'socks4a', 'socks5', 'socks5h')
+
+
+def _validate_proxy_url(value):
+    """Validate that *value* is a well-formed proxy URL (scheme + hostname)."""
+    try:
+        parsed = urllib.parse.urlsplit(value)
+    except ValueError:
+        raise serializers.ValidationError(_('Proxy must be an HTTP, HTTPS, or SOCKS URL.')) from None
```

Ruff (0.15.10)

[error] Local variable `_` referenced before assignment

(F823)